test/lib/utils.js

```diff
@@ -0,0 +1,54 @@
+"use strict";
+
+const {Cc, Ci, components} = require("chrome");
+const {stack: Cs} = components;
+
+function do_get_file(path, allowNonexistent) {
+  try {
+    let lf = Cc["@mozilla.org/file/directory_service;1"]
+      .getService(Ci.nsIProperties)
+      .get("CurWorkD", Ci.nsILocalFile);
+
+    let bits = path.split("/");
+    for (let i = 0; i < bits.length; i++) {
+      if (bits[i]) {
+        if (bits[i] == "..")
+          lf = lf.parent;
+        else
+          lf.append(bits[i]);
+      }
+    }
+
+    if (!allowNonexistent && !lf.exists()) {
+      // Not using do_throw(): caller will continue.
+      let stack = Cs.caller;
+      console.error(`[${stack.name} : ${stack.lineNumber}] ${lf.path} does not exist`);
+    }
+
+    return lf;
+  }
+  catch (ex) {
+    do_throw(ex.toString(), Cs.caller);
+  }
+
+  return null;
+}
+
+function do_throw(error, stack) {
```

Comment:
Nit: rename doThrow(). Underscores in function names is not very JS (and will upset JSCS).